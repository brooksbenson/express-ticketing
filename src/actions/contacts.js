import db from '../firebase/firebase';
import { unsetActiveContact } from './active-contact';

export const addContact = ({ key, ...contact }) => ({
  type: 'ADD_CONTACT',
  key,
  contact
});

export const startAddContact = contact => {
  return async (dispatch, getState) => {
    const { activeAccountKey } = getState();
    const { key } = await db
      .ref(`contact_data/${activeAccountKey}`)
      .push(contact);
    dispatch(addContact({ key, ...contact }));
    return key;
  };
};

export const updateContact = ({ key, ...update }) => ({
  type: 'UPDATE_CONTACT',
  key,
  update
});

export const startUpdateContact = update => {
  return async (dispatch, getState) => {
    const { activeAccountKey, activeContactKey } = getState();
    await db
      .ref(`contact_data/${activeAccountKey}/${activeContactKey}`)
      .set(update);
    dispatch(updateContact({ key: activeContactKey, ...update }));
  };
};

export const setContacts = contacts => ({
  type: 'SET_CONTACTS',
  contacts
});

export const startSetContacts = () => {
  return async (dispatch, getState) => {
    const { activeAccountKey } = getState();
    const snap = await db.ref(`contact_data/${activeAccountKey}`).once('value');
    dispatch(setContacts(snap.val()));
  };
};

export const unsetContacts = () => ({
  type: 'UNSET_CONTACTS'
});

export const startUnsetContacts = () => {
  return dispatch => {
    dispatch(unsetActiveContact());
    dispatch(unsetContacts());
  };
};
