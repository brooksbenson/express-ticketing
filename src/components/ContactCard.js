import React from 'react';

export default props => (
  <div className={props.className}>
    <h3> Contact </h3>
    <ul>
      <li>{props.name}</li>
      <li>{props.email}</li>
      <li>{props.number}</li>
    </ul>
  </div>
);
