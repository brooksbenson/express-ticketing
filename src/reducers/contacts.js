export default (state = null, action) => {
  switch (action.type) {
    case 'ADD_CONTACT':
      return { ...state, [action.key]: action.contact };
    case 'SET_CONTACTS':
      return action.contacts;
    case 'UPDATE_CONTACT':
      return { ...state, [action.key]: action.update };
    default:
      return state;
  }
};
