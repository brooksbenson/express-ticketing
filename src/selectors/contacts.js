export default (contacts, search) => {
  const regex = new RegExp(search.split('').join('.*?'), 'i');
  return contacts.filter(
    c => regex.test(c.name) || regex.test(c.email) || regex.test(c.number)
  );
};
