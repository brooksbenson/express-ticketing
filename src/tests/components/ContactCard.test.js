import React from 'react';
import { shallow } from 'enzyme';
import { contactsArr } from './../fixtures/contacts';
import ContactCard from '../../components/ContactCard';

const { key, ...contact } = contactsArr[0];

test('ContactCard should match snapshot', () => {
  const wrapper = shallow(<ContactCard {...contact} />);
  expect(wrapper).toMatchSnapshot();
});

test('ContactCard should assign wrapper element the class provided by the className prop', () => {
  const wrapper = shallow(
    <ContactCard {...contact} className="contact-card" />
  );
  expect(wrapper.hasClass('contact-card')).toBeTruthy();
});

test('ContactCard should correctly render contact data', () => {
  const wrapper = shallow(<ContactCard {...contact} />);
  const listItemNodes = wrapper.find('li');
  expect(listItemNodes.at(0).text()).toBe(contact.name);
  expect(listItemNodes.at(1).text()).toBe(contact.email);
  expect(listItemNodes.at(2).text()).toBe(contact.number);
});
