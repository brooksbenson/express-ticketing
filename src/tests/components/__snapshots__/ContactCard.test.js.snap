// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactCard should match snapshot 1`] = `
<div>
  <h3>
     Contact 
  </h3>
  <ul>
    <li>
      Jerry Fisher
    </li>
    <li>
      jerry@mail.com
    </li>
    <li>
      8015678989
    </li>
  </ul>
</div>
`;
